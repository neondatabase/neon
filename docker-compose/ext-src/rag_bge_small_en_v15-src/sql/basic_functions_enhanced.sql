-- Basic function tests for chunks_by_token_count
SELECT rag_bge_small_en_v15.chunks_by_token_count('the cat sat on the mat', 3, 2);

SELECT rag_bge_small_en_v15.chunks_by_token_count('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.', 5, 2);

SELECT (rag_bge_small_en_v15.chunks_by_token_count('the cat', 5, 0))[1];

SELECT rag_bge_small_en_v15.chunks_by_token_count('', 5, 2);

SELECT rag_bge_small_en_v15.chunks_by_token_count('a b c d e f g h i j k l m n o p', 3, 1);
