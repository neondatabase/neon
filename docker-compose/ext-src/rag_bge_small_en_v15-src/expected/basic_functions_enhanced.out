-- Basic function tests for chunks_by_token_count
SELECT rag_bge_small_en_v15.chunks_by_token_count('the cat sat on the mat', 3, 2);
                 chunks_by_token_count                  
--------------------------------------------------------
 {"the cat sat","cat sat on","sat on the","on the mat"}
(1 row)

SELECT rag_bge_small_en_v15.chunks_by_token_count('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.', 5, 2);
                                                                              chunks_by_token_count                                                                              
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"Lorem ipsum","ipsum dolor sit","sit amet,",consectetur,"adipiscing elit",elit.,"Sed do","do eiusmod",tempor,"incididunt ut","ut labore et","et dolore magna","magna aliqua."}
(1 row)

SELECT (rag_bge_small_en_v15.chunks_by_token_count('the cat', 5, 0))[1];
 chunks_by_token_count 
-----------------------
 the cat
(1 row)

SELECT rag_bge_small_en_v15.chunks_by_token_count('', 5, 2);
 chunks_by_token_count 
-----------------------
 {}
(1 row)

SELECT rag_bge_small_en_v15.chunks_by_token_count('a b c d e f g h i j k l m n o p', 3, 1);
                      chunks_by_token_count                      
-----------------------------------------------------------------
 {"a b c","c d e","e f g","g h i","i j k","k l m","m n o","o p"}
(1 row)

